janusgraph-test/src/main/java/org/janusgraph/blueprints/AbstractJanusGraphProvider.java

```diff
@@ -134,7 +139,11 @@ public void clear(Graph g, final Configuration configuration) throws Exception {
             JanusGraph graph = (JanusGraph) g;
             if (graph.isOpen()) {
                 if (g.tx().isOpen()) g.tx().rollback();
-                g.close();
+                try {
+                    g.close();
+                } catch (IOException | IllegalStateException e) {
+                    logger.warn("Titan graph may not have closed cleanly", e);
```

Comment:
I don't think so, just tx.close() (see StandardJanusGraph.closeInternal()). This was needed to resolve some sporadic test failures caused by uncaught exceptions associated with "transaction already closed". I can look at it again because I agree it would be better not to have to do this but note this class is in `janusgraph-test` (e.g. it's only used in unit tests).